Branch-SDK/src/io/branch/referral/ServerResponse.java

```diff
@@ -135,4 +140,28 @@ public JSONArray getArray() {
 	public BranchLinkData getLinkData() {
 		return linkData_;
 	}
+
+	/**
+	 * <p>Sets the request object caused this response, as a generic {@link Object}
+	 * instance. This object can be type-cast by other methods within this class:</p>
+	 *
+	 * @param request Generic {@link Object} instance containing the request data caused this response.
+	 */
+	public void setRequestObject(Object request) {
+		this.requestObject_ = request;
+	}
+
+	/**
+	 * <p>Checks whether the request params cause this response is an instance of a
+	 * {@link JSONObject} object, and if so type-casts it to the corresponding class type.</p>
+	 *
+	 * @return A {@link JSONObject} containing the post params sent with corresponding  server request, or a null
+	 * value if the post params is not of the {@link JSONObject} type.
+	 */
+	public JSONObject getRequestObject() {
```

Comment:
Yes I am adding the request Json to the response. The reason for doing that is server is providing enough info on some response. For e.g.   `redeemCredit` succes  response doesn't  have the amount of credit redeemed.It may not be a good idea to keep the parameters in the request locally since it can be overwritten with a successive call of same request.      
  This change actually make the response Object having the  to request parameters with them.  
What you think.Is there a suggestion for a better approach to this issue?